ethcore/src/client/client.rs

```diff
@@ -1327,37 +1327,62 @@ impl BlockChainReset for Client {
 	fn reset(&self, num: u32) -> Result<(), String> {
 		if num as u64 > self.pruning_history() {
 			return Err("Attempting to reset to block with pruned state".into())
+		} else if num == 0 {
+			return Err("invalid number of blocks to reset".into())
 		}
 
-		let (blocks_to_delete, best_block_hash) = self.chain.read()
-			.block_headers_from_best_block(num)
-			.ok_or("Attempted to reset past genesis block")?;
+		let mut blocks_to_delete = Vec::with_capacity(num as usize);
+		let mut best_block_hash = self.chain.read().best_block_hash();
 
-		let mut db_transaction = DBTransaction::with_capacity((num + 1) as usize);
-
-		for hash in &blocks_to_delete {
-			db_transaction.delete(::db::COL_HEADERS, &hash.hash());
-			db_transaction.delete(::db::COL_BODIES, &hash.hash());
-			db_transaction.delete(::db::COL_EXTRA, &hash.hash());
-			Writable::delete::<H256, BlockNumberKey>
-				(&mut db_transaction, ::db::COL_EXTRA, &hash.number());
+		for _ in 0..=num {
```

Comment:
this was previously `num`, not `=num`, could you elaborate why?

Reply:
Yeah, seems that we might remove too much, no? If I want to remove `num=1` block it should only add `best_block_hash` to the list of blocks to delete, so the loop should spin only once.

Reply:
ok so during testing best_block + num - 1 block wasn't being reset.

I later figured out that the test client adds a genesis block. so reverted it